Build error with Midea IR "Follow Me" feature: undefined reference to MIN_TEMP_F and MAX_TEMP_F

### The problem

I'm encountering a build issue when trying to use the Follow Me feature for a Midea air conditioner in ESPHome. 



### Which version of ESPHome has the issue?

2025.7.1

### What type of installation are you using?

pip

### What platform are you using?

ESP32

### Component causing the issue

midea_ac

### YAML Config

```yaml
climate:
  - platform: midea
    id: midea_climate
    name: Climate
    period: 1s
    timeout: 2s
    num_attempts: 3
    autoconf: true
    beeper: true
    visual:
      min_temperature: 16 °C
      max_temperature: 30 °C
      temperature_step: 1 °C
    supported_modes:
      - FAN_ONLY
      - HEAT_COOL
      - COOL
      - HEAT
      - DRY
    custom_fan_modes:
      - SILENT
      - TURBO
    supported_presets:
      - ECO
      - BOOST
      - SLEEP
    supported_swing_modes:
      - VERTICAL
      - HORIZONTAL
      - BOTH
    outdoor_temperature:
      name: "Outdoor Temperature"

  - platform: homeassistant
    id: follow_me_temp
    entity_id: $follow_me_temperature_sensor
    filters:
      - throttle: 10s
      - heartbeat: 2min
      - debounce: 1s
    on_value:
      midea_ac.follow_me:
        temperature: !lambda "return x;"
        beeper: false
```

### Anything in the logs that might be useful for us?

```txt
Compiling .pioenvs/air-conditioner/src/main.cpp.o
In file included from src/esphome/components/adc/adc_sensor.h:9,
                 from src/esphome.h:3,
                 from src/main.cpp:3:
/home/brandon/.platformio/packages/framework-arduinoespressif32-libs/esp32c3/include/esp_adc/deprecated/include/esp_adc_cal.h:17:2: warning: #warning "legacy adc calibration driver is deprecated, please migrate to use esp_adc/adc_cali.h and esp_adc/adc_cali_scheme.h" [-Wcpp]
   17 | #warning "legacy adc calibration driver is deprecated, please migrate to use esp_adc/adc_cali.h and esp_adc/adc_cali_scheme.h"
      |  ^~~~~~~
In file included from src/esphome/components/adc/adc_sensor.h:10:
/home/brandon/.platformio/packages/framework-arduinoespressif32-libs/esp32c3/include/driver/deprecated/driver/adc.h:19:2: warning: #warning "legacy adc driver is deprecated, please migrate to use esp_adc/adc_oneshot.h and esp_adc/adc_continuous.h for oneshot mode and continuous mode drivers respectively" [-Wcpp]
   19 | #warning "legacy adc driver is deprecated, please migrate to use esp_adc/adc_oneshot.h and esp_adc/adc_continuous.h for oneshot mode and continuous mode drivers respectively"
      |  ^~~~~~~
Linking .pioenvs/air-conditioner/firmware.elf
/home/brandon/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/13.2.0/../../../../riscv32-esp-elf/bin/ld: .pioenvs/air-conditioner/src/esphome/components/midea/air_conditioner.cpp.o: in function `esphome::midea::IrFollowMeData::IrFollowMeData(unsigned char, bool, bool)':
/home/brandon/workspaces/esphome/.esphome/build/air-conditioner/src/esphome/components/midea/ir_transmitter.h:28:(.text._ZN7esphome5midea14IrFollowMeDataC2Ehbb[_ZN7esphome5midea14IrFollowMeDataC5Ehbb]+0x8e): undefined reference to `esphome::midea::IrFollowMeData::MIN_TEMP_F'
/home/brandon/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/13.2.0/../../../../riscv32-esp-elf/bin/ld: .pioenvs/air-conditioner/src/esphome/components/midea/air_conditioner.cpp.o: in function `unsigned char const& std::min<unsigned char>(unsigned char const&, unsigned char const&)':
/home/brandon/.platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/13.2.0/bits/stl_algobase.h:240:(.text._ZN7esphome5midea14IrFollowMeDataC2Ehbb[_ZN7esphome5midea14IrFollowMeDataC5Ehbb]+0x92): undefined reference to `esphome::midea::IrFollowMeData::MIN_TEMP_F'
/home/brandon/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/13.2.0/../../../../riscv32-esp-elf/bin/ld: /home/brandon/.platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/13.2.0/bits/stl_algobase.h:240:(.text._ZN7esphome5midea14IrFollowMeDataC2Ehbb[_ZN7esphome5midea14IrFollowMeDataC5Ehbb]+0x98): undefined reference to `esphome::midea::IrFollowMeData::MAX_TEMP_F'
/home/brandon/.platformio/packages/toolchain-riscv32-esp/bin/../lib/gcc/riscv32-esp-elf/13.2.0/../../../../riscv32-esp-elf/bin/ld: /home/brandon/.platformio/packages/toolchain-riscv32-esp/riscv32-esp-elf/include/c++/13.2.0/bits/stl_algobase.h:239:(.text._ZN7esphome5midea14IrFollowMeDataC2Ehbb[_ZN7esphome5midea14IrFollowMeDataC5Ehbb]+0x9c): undefined reference to `esphome::midea::IrFollowMeData::MAX_TEMP_F'
collect2: error: ld returned 1 exit status
*** [.pioenvs/air-conditioner/firmware.elf] Error 1
```

### Additional information

_No response_

I've the same issue.

```bash
/data/build/<device_name>/src/esphome/components/midea/ir_transmitter.h:28:(.text._ZN7esphome5midea14IrFollowMeDataC2Ehbb[_ZN7esphome5midea14IrFollowMeDataC5Ehbb]+0x8e): undefined reference to `esphome::midea::IrFollowMeData::MIN_TEMP_F'
```

--- a/esphome/components/midea/ir_transmitter.h
+++ b/esphome/components/midea/ir_transmitter.h
@@ -54,15 +54,15 @@ class IrFollowMeData : public IrData {
   void set_fahrenheit(bool val) { this->set_mask_(2, val, 32); }
 
  protected:
-  static const uint8_t MIN_TEMP_C = 0;
-  static const uint8_t MAX_TEMP_C = 37;
+  inline static constexpr uint8_t MIN_TEMP_C = 0;
+  inline static constexpr uint8_t MAX_TEMP_C = 37;
 
   // see
   // https://github.com/crankyoldgit/IRremoteESP8266/blob/9bdf8abcb465268c5409db99dc83a26df64c7445/src/ir_Midea.h#L116
-  static const uint8_t MIN_TEMP_F = 32;
+  inline static constexpr uint8_t MIN_TEMP_F = 32;
   // see
   // https://github.com/crankyoldgit/IRremoteESP8266/blob/9bdf8abcb465268c5409db99dc83a26df64c7445/src/ir_Midea.h#L117
-  static const uint8_t MAX_TEMP_F = 99;
+  inline static constexpr uint8_t MAX_TEMP_F = 99;
 };
 
 class IrSpecialData : public IrData {

